cmd/utils.go

```diff
@@ -349,3 +350,69 @@ func parseAttribute(attrs string) (map[string]string, error) {
 
 	return attribute, nil
 }
+
+// EqualEvents tells whether a and b contain the same events
+func EqualEvents(a, b []minio.NotificationEventType) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	sort.SliceStable(a, func(i, j int) bool {
+		return a[i] < a[j]
+	})
+	sort.SliceStable(b, func(i, j int) bool {
+		return b[i] < b[j]
+	})
+	for i, v := range a {
+		if v != b[i] {
+			return false
+		}
+	}
+	return true
+}
+
+// EqualFilterRules tells whether a and b contain the same filters
+func EqualFilterRules(a, b []minio.FilterRule) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	sort.SliceStable(a, func(i, j int) bool {
+		left := fmt.Sprintf("%s-%s", a[i].Name, a[i].Value)
+		right := fmt.Sprintf("%s-%s", a[j].Name, a[j].Value)
+		return left < right
+	})
+	sort.SliceStable(b, func(i, j int) bool {
+		left := fmt.Sprintf("%s-%s", b[i].Name, b[i].Value)
+		right := fmt.Sprintf("%s-%s", b[j].Name, b[j].Value)
+		return left < right
+	})
+
+	for i, v := range a {
+		if v != b[i] {
+			return false
+		}
+	}
+	return true
+}
+
+// EqualNotificationConfig returns wether two `minio.NotificationConfig` are equivalent
+func EqualNotificationConfig(v minio.NotificationConfig, eventsTyped []minio.NotificationEventType, prefix, suffix string) bool {
+	//Compare events
+	passEvents := false
+	if len(v.Events) == len(eventsTyped) && EqualEvents(v.Events, eventsTyped) {
```

Comment:
We can remove `len(v.Events) == len(eventsTyped)` check since it is done inside `EqualEvents`.

Reply:
fixed!